iGCS/Models/GCSMissionItemModel.m

```diff
@@ -0,0 +1,121 @@
+//
+//  GCSMissionItemModel.m
+//  iGCS
+//
+//  Created by BRYAN GALUSHA on 2/16/15.
+//
+//
+
+#import "GCSMissionItemModel.h"
+
+
+@implementation GCSMissionItemModel
+
+
+/*
+float param1; ///< PARAM1, see MAV_CMD enum
+float param2; ///< PARAM2, see MAV_CMD enum
+float param3; ///< PARAM3, see MAV_CMD enum
+float param4; ///< PARAM4, see MAV_CMD enum
+float x; ///< PARAM5 / local: x position, global: latitude
+float y; ///< PARAM6 / y position: global: longitude
+float z; ///< PARAM7 / z position: global: altitude (relative or absolute, depending on frame.
+uint16_t seq; ///< Sequence
+uint16_t command; ///< The scheduled action for the MISSION. see MAV_CMD in common.xml MAVLink specs
+uint8_t target_system; ///< System ID
+uint8_t target_component; ///< Component ID
+uint8_t frame; ///< The coordinate system of the MISSION. see MAV_FRAME in mavlink_types.h
+uint8_t current; ///< false:0, true:1
+uint8_t autocontinue; ///< autocontinue to next wp
+} mavlink_mission_item_t;
+ */
+
+-(void)missionItemFromMavlink:(mavlink_mission_item_t) mavMissionItem {
+    _param1 = [NSNumber numberWithFloat:mavMissionItem.param1];
+    _param2 = [NSNumber numberWithFloat:mavMissionItem.param2];
+    _param3 = [NSNumber numberWithFloat:mavMissionItem.param3];
+    _param4 = [NSNumber numberWithFloat:mavMissionItem.param4];
+    _x = [NSNumber numberWithFloat:mavMissionItem.x];
+    _y = [NSNumber numberWithFloat:mavMissionItem.y];
+    _z = [NSNumber numberWithFloat:mavMissionItem.z];
+    _seq = [NSNumber numberWithUnsignedInteger:mavMissionItem.seq];
+    _command = [NSNumber numberWithUnsignedInteger:mavMissionItem.command];
+    _target_system = [NSNumber numberWithUnsignedInteger:mavMissionItem.target_system];
+    _target_component = [NSNumber numberWithUnsignedInteger:mavMissionItem.target_component];
+    _frame = [NSNumber numberWithUnsignedInteger:mavMissionItem.frame];
+    _current = [NSNumber numberWithBool:mavMissionItem.current];
+    _autocontinue = [NSNumber numberWithUnsignedInteger:mavMissionItem.autocontinue];
+        
+}
+
+
+-(mavlink_mission_item_t)missionItemToMavlink {
+    mavlink_mission_item_t missionItem;
+    missionItem.param1 = [_param1 floatValue];
```

Comment:
Have resolved to use properties instead of instance variables throughout the code base for things like this.